core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java

```diff
@@ -34,172 +24,172 @@
-      Runtime runtime = null;
-      try {
-        libc = LibraryLoader.create(LibC.class).load("c");
-        runtime = Runtime.getRuntime(libc);
-      } catch (Throwable t) {
-        libc = null;
-        LOG.debug("Error loading libc", t);
-      }
-      LIB_C = libc;
-      LIB_C_RUNTIME = runtime;
-      boolean getTimeOfDayAvailable = false;
-      if (LIB_C_RUNTIME != null) {
-        try {
-          getTimeOfDayAvailable = LIB_C.gettimeofday(new Timeval(LIB_C_RUNTIME), null) == 0;
-        } catch (Throwable t) {
-          LOG.debug("Error accessing libc.gettimeofday()", t);
-        }
-      }
-      GET_TIME_OF_DAY_AVAILABLE = getTimeOfDayAvailable;
-    }
-  }
+    /** 64 bit Power PC little endian */
+    PPC64LE,
 
-  /** @see LibCLoader */
-  private static class PosixLoader {
-    @SuppressWarnings("VariableNameSameAsType")
-    private static final jnr.posix.POSIX POSIX;
+    /** 32 bit Sun sparc */
+    SPARC,
 
-    private static final boolean GET_PID_AVAILABLE;
+    /** 64 bit Sun sparc */
+    SPARCV9,
 
-    static {
-      jnr.posix.POSIX posix;
-      try {
-        posix = POSIXFactory.getPOSIX(new DefaultPOSIXHandler(), true);
-      } catch (Throwable t) {
-        posix = null;
-        LOG.debug("Error loading POSIX", t);
-      }
-      POSIX = posix;
-      boolean getPidAvailable = false;
-      if (POSIX != null) {
-        try {
-          POSIX.getpid();
-          getPidAvailable = true;
-        } catch (Throwable t) {
-          LOG.debug("Error accessing posix.getpid()", t);
-        }
-      }
-      GET_PID_AVAILABLE = getPidAvailable;
-    }
-  }
+    /** IBM zSeries S/390 */
+    S390X,
 
-  private static class PlatformLoader {
+    /** 32 bit MIPS (used by nestedvm) */
+    MIPS32,
 
-    private static final Platform PLATFORM;
+    /** 32 bit ARM */
+    ARM,
 
-    static {
-      Platform platform = null;
-      try {
-        Class<?> platformClass = Reflection.loadClass(null, "jnr.ffi.Platform");
-        if (platformClass != null) {
-          Method getNativePlatform = platformClass.getMethod("getNativePlatform");
-          platform = (Platform) getNativePlatform.invoke(null);
-        }
-      } catch (Throwable t) {
-        LOG.debug("Error loading jnr.ffi.Platform class, this class will not be available.", t);
+    /** 64 bit ARM */
+    AARCH64,
+
+    /**
+     * Unknown CPU architecture. A best effort will be made to infer architecture specific values
+     * such as address and long size.
+     */
+    UNKNOWN;
+  }
+
+  private static CPU determineCPU() {
+    String archString = System.getProperty("os.arch");
+    assert (archString != null);
+    assert (archString.toUpperCase(LOCALE) != null);
+    assert (archString.toLowerCase(LOCALE) != null);
+    if (equalsIgnoreCase("x86", archString)
+        || equalsIgnoreCase("i386", archString)
+        || equalsIgnoreCase("i86pc", archString)
+        || equalsIgnoreCase("i686", archString)) {
+      return CPU.I386;
```

Comment:
The enum should be called `Cpu` in accordance to our [naming conventions](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case), so that will indirectly address this problem.

We should also rename `getCPU` to `getCpu`.

Reply:
The Cpu name across the board does seem more sensible... I'll update this as well.